core/drivers/firewall/stm32_rifsc.c

```diff
@@ -495,17 +487,19 @@ static TEE_Result stm32_rimu_setup(struct rifsc_platdata *pdata)
 	return TEE_SUCCESS;
 }
 
-#if defined(CFG_STM32MP25)
 static TEE_Result stm32_risal_cfg(struct rifsc_platdata *pdata,
 				  struct risal_cfg *risal)
 {
 	struct rifsc_driver_data *drv_data = pdata->drv_data;
-	uintptr_t offset_a = _RIFSC_RISAL_CFGR0_A(risal->id);
-	uintptr_t offset_b = _RIFSC_RISAL_CFGR0_B(risal->id);
+	uintptr_t offset_a = 0;
+	uintptr_t offset_b = 0;
 
 	if (!risal || risal->id > drv_data->nb_risal)
 		return TEE_ERROR_BAD_PARAMETERS;
 
+	offset_a = _RIFSC_RISAL_CFGR0_A(risal->id);
+	offset_b = _RIFSC_RISAL_CFGR0_B(risal->id);
+
 	if (drv_data->rif_en) {
```

Comment:
For consistency, maybe:
```c
	if (drv_data->rif_en) {
		uintptr_t offset_a = _RIFSC_RISAL_CFGR0_A(risal->id);
		uintptr_t offset_b = _RIFSC_RISAL_CFGR0_B(risal->id);

		...
```
Note that we prefer `vaddr_t` instead of `uintptr_t` but maybe not worth changing this source file that already uses the later at many places.